clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/modules/CloudLoggingModule.java

```diff
@@ -19,28 +19,58 @@
 import com.abcxyz.lumberjack.auditlogclient.config.AuditLoggingConfiguration;
 import com.abcxyz.lumberjack.auditlogclient.utils.ConfigUtils;
 import com.google.api.client.util.Strings;
+import com.google.api.gax.batching.BatchingSettings;
+import com.google.cloud.logging.LogEntry;
 import com.google.cloud.logging.Logging;
 import com.google.cloud.logging.LoggingOptions;
+import com.google.cloud.logging.Payload.StringPayload;
+import com.google.cloud.logging.Severity;
 import com.google.cloud.logging.Synchronicity;
 import com.google.inject.AbstractModule;
 import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import java.util.Collections;
+import lombok.extern.slf4j.Slf4j;
+import org.threeten.bp.Duration;
 
+@Slf4j
 public class CloudLoggingModule extends AbstractModule {
   @Provides
+  @Singleton
   Logging logging(AuditLoggingConfiguration configuration) {
     LoggingOptions.Builder loggingOptionsBuilder = LoggingOptions.newBuilder();
     if (configuration.getBackend().cloudLoggingEnabled()
         && !Strings.isNullOrEmpty(configuration.getBackend().getCloudlogging().getProject())) {
       if (configuration.getBackend().getCloudlogging().useDefaultProject()) {
         throw new IllegalStateException("Cannot set cloud logging project if default is enabled.");
       }
-      loggingOptionsBuilder.setProjectId(configuration.getBackend().getCloudlogging().getProject());
+      loggingOptionsBuilder
+          .setProjectId(configuration.getBackend().getCloudlogging().getProject())
+          .setBatchingSettings(
+              BatchingSettings.newBuilder()
+                  .setElementCountThreshold(10L)
+                  .setDelayThreshold(Duration.ofSeconds(1L))
+                  // LogEntry size limit ref: https://cloud.google.com/logging/quotas.
+                  .setRequestByteThreshold(2560000L)
+                  .build());
     }
     Logging logging = loggingOptionsBuilder.build().getService();
     if (ConfigUtils.shouldFailClose(configuration.getLogMode())) {
       logging.setWriteSynchronicity(Synchronicity.SYNC);
     } else {
       logging.setWriteSynchronicity(Synchronicity.ASYNC);
+      // warmup request as a temporary workaround until this fixed:
+      // https://github.com/googleapis/java-logging/issues/1469.
```

Comment:
This is super weird if true. Are you able to reproduce the problem locally?

Reply:
Seems due to cloud run cold start, will look into a fix.

Reply:
fixed.